velox/exec/VectorHasher.h

```diff
@@ -172,6 +172,11 @@ class VectorHasher {
   // computeValueIds(). The decoded vector can be accessed via decodedVector()
   // getter.
   void decode(const BaseVector& vector, const SelectivityVector& rows) {
+    VELOX_CHECK(
+        type_->kindEquals(vector.type()),
```

Comment:
would any properties of the logical type matter here? like if the precision of the decimal types are different?